middleware/middleware.go

```diff
@@ -12,12 +12,14 @@ import (
 	"github.com/chihaya/chihaya/storage"
 )
 
+// Config holds the configuration of a tracker's Logic.
 type Config struct {
 	AnnounceInterval time.Duration `yaml:"announce_interval"`
 }
 
 var _ frontend.TrackerLogic = &Logic{}
 
+// NewLogic creates a new Logic from the given parameters.
 func NewLogic(cfg Config, peerStore storage.PeerStore, preHooks, postHooks []Hook) *Logic {
```

Comment:
NewLogic creates a new instance of a TrackerLogic that executes the provided middleware hooks.